src/libs/Xinterface/QuestFileReader/QuestFileReader.cpp

```diff
@@ -1,644 +1,392 @@
-                GetSubStringFromString(tmpstr, buf, 256);
-                const auto *const retVal = ptr;
-                // find the end of the quest title
-                while (ptr != nullptr)
-                {
-                    tmpstr = ptr;
-                    tokType = GetToken(tmpstr);
-                    if (tokType == TOKEN_QTEXT || tokType == TOKEN_QUEST)
-                        break;
-                    GetNextString(ptr);
-                    if (tmpstr == ptr)
-                        break;
-                }
-                if (ptr != tmpstr)
-                    slen = ptr - retVal;
-                else if (retVal != nullptr)
-                    slen = strlen(retVal);
-                else
-                    slen = 0;
-                if (retVal != nullptr)
-                    return retVal;
-            }
-        }
-        else
-            break;
-    }
-    buf[0] = 0;
-    slen = 0;
-    return nullptr;
-}
 
-QUEST_FILE_READER::QUEST_FILE_READER()
-{
-    m_pFileBuf = nullptr;
-}
+    it = std::next(textEnd);
 
-QUEST_FILE_READER::~QUEST_FILE_READER()
-{
-    CloseQuestsQuery();
-}
+    return {std::string_view(textStart, textEnd), patternStart};
+};
 
-bool QUEST_FILE_READER::InitQuestsQuery()
-{
-    CloseQuestsQuery();
+} // namespace
 
-    if (m_aQuest.size() == 0)
-    {
-        for (long n = 0; n < m_aQuestFileName.size(); n++)
-        {
-            auto fileS = fio->_CreateFile(m_aQuestFileName[n].c_str(), std::ios::binary | std::ios::in);
-            if (!fileS.is_open())
-            {
-                core.Trace("WARNING! Can`t open quest log file %s", m_aQuestFileName[n].c_str());
-                continue;
-            }
-
-            const auto filesize = fio->_GetFileSize(m_aQuestFileName[n].c_str());
-            if (filesize == 0)
-            {
-                core.Trace("Empty quest log file %s", m_aQuestFileName[n].c_str());
-                fio->_CloseFile(fileS);
-                continue;
-            }
-
-            long foffset = 0;
-            if (m_pFileBuf == nullptr)
-            {
-                m_pFileBuf = static_cast<char *>(malloc(filesize + 1));
-            }
-            else
-            {
-                foffset = strlen(m_pFileBuf);
-                m_pFileBuf = static_cast<char *>(realloc(m_pFileBuf, foffset + filesize + 1));
-            }
-
-            if (m_pFileBuf == nullptr)
-            {
-                throw std::runtime_error("allocate memory error");
-            }
-
-            if (!fio->_ReadFile(fileS, &m_pFileBuf[foffset], filesize))
-            {
-                core.Trace("Can`t read quest log file: %s", m_aQuestFileName[n].c_str());
-            }
-            fio->_CloseFile(fileS);
-            m_pFileBuf[foffset + filesize] = 0;
-        }
-    }
-
-    if (m_pFileBuf == nullptr || m_pFileBuf[0] == 0)
-    {
-        return false;
-    }
-    return true;
+namespace storm
+{
+QuestFileReader::QuestFileReader()
```

Comment:
`QuestFileReader` constructor and destructor are no longer needed. So you can remove them if you want.